app/models/manageiq/providers/amazon/manager_mixin.rb

```diff
@@ -48,19 +48,23 @@ module ClassMethods
     # Connections
     #
 
-    def raw_connect(access_key_id, secret_access_key, service, region, proxy_uri = nil, validate = false)
+    def raw_connect(access_key_id, secret_access_key, service, region, proxy_uri = nil, validate = false, uri = nil)
       require 'aws-sdk'
       require 'patches/aws-sdk-core/seahorse_client_net_http_pool_patch'
 
-      connection = Aws.const_get(service)::Resource.new(
+      options = {
         :access_key_id     => access_key_id,
         :secret_access_key => MiqPassword.try_decrypt(secret_access_key),
         :region            => region,
         :http_proxy        => proxy_uri,
         :logger            => $aws_log,
         :log_level         => :debug,
         :log_formatter     => Aws::Log::Formatter.new(Aws::Log::Formatter.default.pattern.chomp)
-      )
+      }
+
+      options[:endpoint] = uri.to_s if uri && uri.to_s.present?
+
```

Comment:
@djberg96 -`present` tests for nil too so there is no need to explicitly check it.